main_adapter.py

```diff
@@ -12,6 +12,12 @@ class ModelAdapter(dl.BaseModelAdapter):
     def __init__(self, model_entity: dl.Model):
         if os.environ.get("NVIDIA_NIM_API_KEY", None) is None:
             raise ValueError(f"Missing API key: NVIDIA_NIM_API_KEY")
+        self.api_key = os.environ.get("NVIDIA_NIM_API_KEY", None)
```

Comment:
I always question myself about this: is it better to get the api key just once at service initialization or before every request we do?

Reply:
here it is used in multiple places, so to avoid the code duplication i moved it to init